qcodes/config/station_configurator.py

```diff
@@ -0,0 +1,249 @@
+from contextlib import suppress
+from typing import Optional
+from functools import partial
+import importlib
+import logging
+import warnings
+import os
+from types import MethodType
+from copy import deepcopy
+import qcodes
+from qcodes.instrument.base import Instrument
+from qcodes.station import Station
+import qcodes.utils.validators as validators
+from qcodes.instrument.parameter import Parameter
+from qcodes.monitor.monitor import Monitor
+from qcodes.instrument.parameter import DelegateParameter
+use_pyyaml = False
+try:
+    from ruamel.yaml import YAML
+except ImportError:
+    use_pyyaml = True
+    warnings.warn(
+        "ruamel yaml not found station configurator is falling back to pyyaml. "
+        "It's highly recommended to install ruamel.yaml. This fixes issues with "
+        "scientific notation and duplicate instruments in the YAML file")
+
+log = logging.getLogger(__name__)
+
+# config from the qcodesrc.json file (working directory, home or qcodes dir)
+enable_forced_reconnect = qcodes.config["station_configurator"]["enable_forced_reconnect"]
+default_folder = qcodes.config["station_configurator"]["default_folder"]
+default_file = qcodes.config["station_configurator"]["default_file"]
+
+
+class StationConfigurator:
+    """
+    The StationConfig class enables the easy creation of intstruments from a
+    yaml file.
+    Example:
+    >>> scfg = StationConfig('exampleConfig.yaml')
+    >>> dmm1 = scfg.create_instrument('dmm1')
+    """
+
+    PARAMETER_ATTRIBUTES = ['label', 'unit', 'scale', 'inter_delay', 'delay',
+                            'step', 'offset']
+
+    def __init__(self, filename: Optional[str] = None,
+                 station: Optional[Station] = None) -> None:
+        self.monitor_parameters = {}
+
+        if station is None:
+            station = Station.default or Station()
+        self.station = station
+        self.filename = filename
+        # a list of method names that got added by :meth:`load_file`
+        self._added_methods: List[str] = []
+
+        self.load_file(self.filename)
+
+    def load_file(self, filename: Optional[str] = None):
+
+        # 1. load config from file
+        if use_pyyaml:
```

Comment:
Is there a reason to support usage of pyyaml? Seems like a good time to choose using ruamel (at least I think it is just better) instead of having two versions of dependencies spread in lab.

Reply:
Yes there is unfortunately. There has been an issue with writing back the meta data with ruamel.. I still haven't found out why this happens but I am happy they have pyaml.